### PR TITLE
Updates to zipkin2 json format

### LINKED ISSUE #68: Any plans to update this to zipkin v2 api?

### PR DESCRIPTION
**Pull request by adriancole**
This needs a decision.. For example, do we change the package name, or
just break api?

The version is snapshot due to some validation we didn't do until https://github.com/openzipkin/zipkin/pull/2101

Fixes #68

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -41,6 +41,7 @@ allprojects {
     }
 
     repositories {
+        mavenLocal()
```
Comment (bhumit) [inline build.gradle]: you might have accidentally missed out on removing this :)

**Comment by nicmunroe** [comment]
@adriancole I'm thinking we should add a new module, maybe `wingtips-zipkin2`? That way users could continue to use the existing v1 module without breakage, users who want to use v2 would just be a simple dependency change, and going forward if someone needs to export to zipkin v1 for some reason they still can.

**Comment by adriancole** [comment]
so the zipkin2 library can also export in v1 format. I am happy to make a
config available  if you don't have opinions.. basically if you look at
SpanBytesEncoder all formats are supported.

Ideal imho is that people stop using the v1 dep as it won't be maintained.
Maybe starting with deprecating in favor of a zipkin2 module which has
instructions for old format could do the trick?

wdyt?

**Comment by nicmunroe** [comment]
Ah, if we can support both v1 and v2 in the same module (without pulling in old zipkin v1 deps), then a config option seems like the correct way to do it.

And deprecating the old module in favor of a new `wingtips-zipkin2` works for me.

**Comment by adriancole** [comment]
I did some renaming etc and also dropped "lc" from the zipkin2 impl as it isn't worth it. Mostly "lc" was around as a hack to give some place to attach a service name. In the v2 model, localEndpoint fixes this.

raising this point separately before doing the v1 format property just in case it raises antibodies

**Comment by nicmunroe** [comment]
No antibodies - I'm always going to defer to your recommendations for how to translate from wingtips spans to zipkin spans, so if you notice anything else that seems wrong please point it out!

File: build.gradle [hunk 2]
```diff
@@ -41,6 +41,8 @@ allprojects {
     }
 
     repositories {
+        // temporary until next zipkin release
+        maven { url 'http://oss.jfrog.org/artifactory/oss-snapshot-local' }
```
Comment (nicmunroe) [inline build.gradle]: What's the timeline on the next zipkin release?

File: wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImpl.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            .name(wingtipsSpan.getSpanName())
+            .parentId(wingtipsSpan.getParentSpanId())
+            .traceId(traceId)
+            .build();
+    }
+
+    protected zipkin2.Span.Builder createNewZipkinSpanBuilderWithSpanPurposeAnnotations(
+        Span wingtipsSpan, long startEpochMicros, long durationMicros, Endpoint zipkinEndpoint
+    ) {
+        zipkin2.Span.Builder zsb = zipkin2.Span.newBuilder()
```
Comment (nicmunroe) [inline wingtips-zipkin2/src/main/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImpl.java]: Now that we don't have to build timestamps for sr/ss/cr/cs annotations, we can move most of this into the main `convertWingtipsSpanToZipkinSpan(...)` method, and just have a simple helper method for converting wingtips `SpanPurpose` -> zipkin `Kind`, yes/no?

You don't need to do that refactoring - I'm planning on doing some other cleanup and refactoring on top of what you've done (e.g. probably moving to supporting `zipkin2.reporter.Reporter`s directly). Just making sure I'm not missing something.

File: wingtips-zipkin2/src/test/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImplTest.java [hunk 4]
```diff
@@ -150,43 +132,40 @@ public void convertWingtipsSpanToZipkinSpan_works_as_expected_for_128_bit_trace_
...
+        assertThat(zipkinSpan.traceId()).isEqualTo(hex128Bits);
     }
 
     @DataProvider(value = {
         "                                      ", // empty trace ID
         "123e4567-e89b-12d3-a456-426655440000  "  // UUID format (hyphens and also >32 chars)
     }, splitBy = "\\|")
     @Test
-    public void convertWingtipsSpanToZipkinSpan_throws_NumberFormatException_for_illegal_args(final String badHexString) {
+    public void convertWingtipsSpanToZipkinSpan_throws_IllegalArgumentException(final String badHexString) {
```
Comment (nicmunroe) [inline wingtips-zipkin2/src/test/java/com/nike/wingtips/zipkin2/util/WingtipsToZipkinSpanConverterDefaultImplTest.java]: The trailing `_for_illegal_args` got cut off the test method name.

**Comment by adriancole** [comment]
Hi, Nic.

FYI the snapshot dep is only about a couple edge cases in your tests which
are not likely to occur. We aren't likely to publish a version until later
this week as in a zipkin pow-wow at the moment and the master has some
unfinished things in it.

Appreciate you taking this on.. if you want to you could put in a temporary
hack to address the couple things that fail if you move back to latest
release, or TODO to un-ignore the test. Sorry lost steam, in the middle of
too much things right now :P

**Comment by nicmunroe** [comment]
No worries, I know how it can be. Thank you for taking it this far - I wasn't expecting you to jump on this and it's much appreciated!

**Comment by adriancole** [comment]
ps zipkin 2.10 is out so that part should be surmountable. Due to a flurry of travel I might have to lean on you to wrap this up unless you are also slammed :)

**Comment by nicmunroe** [comment]
I haven't forgotten about this, just busy. I hope to jump on this in the next day or two. We'll see how it goes.

**Comment by nicmunroe** [comment]
Working on this now.

**Comment by nicmunroe** [comment]
@adriancole I'm not sure if I can push commits to your PR, so I created a new one (#70) that uses your commits and builds on them. If there's a way to add my commits to your PR then give me a quick tutorial and I'll do that, otherwise we can close this PR in favor of #70.

**Comment by adriancole** [comment]
close it.. not a concern worth mucking with :)

**Comment by nicmunroe** [comment]
Right on. Closing in favor of #70.
